Sources/ETCD/ETCDClient.swift

```diff
@@ -120,5 +123,43 @@ public final class EtcdClient: @unchecked Sendable {
     public func put(_ key: String, value: String) async throws {
         try await put(key.utf8, value: value.utf8)
     }
+    
+    public func watch<Result>(_ key: some Sequence<UInt8>, _ operation: (WatchAsyncSequence) async throws -> Result) async throws -> Result {
+           let request = [Etcdserverpb_WatchRequest.with { $0.createRequest.key = Data(key) }]
+           let watchAsyncSequence = WatchAsyncSequence(watchClient.watch(request))
+           return try await operation(watchAsyncSequence)
+    }
+    
+    public func watch<Result>(_ key: String, _ operation: (WatchAsyncSequence) async throws -> Result) async throws -> Result {
+        let request = [Etcdserverpb_WatchRequest.with { $0.createRequest.key = Data(key.utf8) }]
+           let watchAsyncSequence = WatchAsyncSequence(watchClient.watch(request))
+           return try await operation(watchAsyncSequence)
+    }
+}
+
+public struct WatchAsyncSequence: AsyncSequence, AsyncIteratorProtocol {
+    public typealias Element = Data
```

Comment:
`Data` isn't correct here. We need to create a new `WatchEvent` structure that is similar to the `Event` structure in the watch proto.